server/app/controllers/core.controller.js

```diff
@@ -9,9 +10,80 @@ const {
 } = require("../models/core/destination.model")
 
 const F5_HOST_IP = '10.206.40.100';
+const DESTINATION_MAPPING_SCRIPT_PATH = '../scripts/get_all_destination_names.py';
+const DESTINATION_MAPPING_FILE = 'controller_info.json';
+
+const parseDestinationMappingsData = (req, res, destinationScriptResponse) => {
+    const { tenants, clouds } = destinationScriptResponse;
+
+    if(!Array.isArray(tenants) || !Array.isArray(clouds)) {
+        res.status(500).json({ error: 'Something went wrong while parsing destination mapping data.'});
+        return;
+    }
+
+    const destinationMappings = {
+        'tenants': []
+    };
+
+    for (const tenant of tenants) {
+        const formattedTenantObj = {
+            tenant: '',
+            clouds: []
+        };
+        if (tenant) {
+            formattedTenantObj.tenant = tenant;
+            for (const cloud of clouds) {
+                const { vrf, seg, cloud_name } = cloud;
+                if (cloud_name && (Array.isArray(vrf) || Array.isArray(seg))) {
+                    const currentTenantVrf = vrf.filter(item => item.tenant === tenant);
+                    const currentTenantSeg = seg.filter(item => item.tenant === tenant);
+
+                    if (currentTenantVrf.length) {
+                        const cloudToUpdate = formattedTenantObj.clouds.find(cloud => cloud.cloud_name === cloud_name);
+                        if (cloudToUpdate) {
+                            cloudToUpdate.vrf = currentTenantVrf.map(item => {
+                                return { 'name': item.name }
+                            });
+                        } else {
+                            formattedTenantObj.clouds.push({
+                                'cloud_name': cloud_name,
+                                vrf: currentTenantVrf.map(item => {
+                                    return { 'name': item.name }
+                                })
```

Comment:
```suggestion
                                vrf: currentTenantVrf.map(item => { 'name': item.name })
```

Reply:
This throws an error as with this syntax it considers the object as function body. Sticking to old return approach for better readability and maintainability.